exec/java-exec/src/main/java/org/apache/drill/exec/vector/complex/MapVector.java

```diff
@@ -355,17 +353,19 @@ public void generateTestData(int values) { }
 
   @Override
   public void clear() {
-    valueCount = 0;
-    for (ValueVector v : getChildren()) {
+    for (final ValueVector v : getChildren()) {
       v.clear();
     }
+    valueCount = 0;
   }
 
   @Override
   public void close() {
-    for (final ValueVector v : getChildren()) {
+    final Collection<ValueVector> vectors = getChildren();
```

Comment:
super.close()  and resetting value count should be just fine here. remaining seems redundant if I am not missing anything.